examples/multi-functions-synchronization/ai_ao_sync.py

```diff
@@ -0,0 +1,98 @@
+﻿"""Example of analog input and output synchronization.
+
+This example demonstrates how to continuously acquire and
+generate data at the same time, synchronized with one another.
+"""
+
+from typing import Tuple
+
+import numpy as np
+import numpy.typing
+
+import nidaqmx
+from nidaqmx.constants import AcquisitionType
+
+
+def generate_sine_wave(
+    frequency: float,
+    amplitude: float,
+    sampling_rate: float,
+    phase_in: float,
+    number_of_samples: int,
+) -> Tuple[numpy.typing.NDArray[numpy.double], float]:
+    """Generates a sine wave with a specified phase.
+
+    Args:
+        frequency (float): Specifies the frequency of the sine wave.
+        amplitude (float): Specifies the amplitude of the sine wave.
+        sampling_rate (float): Specifies the sampling rate of the sine wave.
+        phase_in (float): Specifies the phase of the sine wave in radians.
+        number_of_samples (int): Specifies the number of samples to generate.
+
+    Returns:
+        Tuple[numpy.typing.NDArray[numpy.double], float]: Indicates a tuple
+        containing the generated data and the phase of the sine wave after generation.
+    """
+    duration_time = number_of_samples / sampling_rate
+    duration_radians = duration_time * 2 * np.pi
+    phase_out = (phase_in + duration_radians) % (2 * np.pi)
+    t = np.linspace(phase_in, phase_in + duration_radians, number_of_samples, endpoint=False)
+
+    return (amplitude * np.sin(frequency * t), phase_out)
+
+
+def main():
+    """Continuously acquires and generate data at the same time."""
+    total_read = 0
+    phase = 0.0
+    number_of_samples = 1000
+    task_ai = nidaqmx.Task()
+    task_ao = nidaqmx.Task()
+
+    def callback(task_handle, every_n_samples_event_type, number_of_samples, callback_data):
+        """Callback function for reading singals."""
+        nonlocal total_read
+        read = task_ai.read(number_of_samples_per_channel=number_of_samples)
+        total_read += len(read)
+        print(f"Acquired data: {len(read)} samples. Total {total_read}.", end="\r")
+
+        return 0
+
+    task_ai.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+    task_ai.timing.cfg_samp_clk_timing(1000.0, sample_mode=AcquisitionType.CONTINUOUS)
+    task_ai.register_every_n_samples_acquired_into_buffer_event(1000, callback)
+    task_ao.ao_channels.add_ao_voltage_chan("Dev1/ao0")
+    task_ao.timing.cfg_samp_clk_timing(1000.0, sample_mode=AcquisitionType.CONTINUOUS)
+    task_ao.triggers.start_trigger.cfg_dig_edge_start_trig("/Dev1/ai/StartTrigger")
```

Comment:
You can reduce duplication by omitting "/Dev1/", but this won't work with cDAQ because the trigger terminal is on the chassis, not the module. That's why the LV and C examples have a "GetTerminalNameWithDevPrefix" function.

Reply:
"/Dev1/" is removed from the examples. It seems like `GetNthTaskDevice` is not yet implemented, it's required for `GetTerminalNameWithDevPrefix`. We can update these examples when `GetNthTaskDevice` API is available.

Reply:
`GetNthTaskDevice` is not required. Use the `Task.devices` property.

```python
for device in task.devices:
    if device.product_category not in [ProductCategory.C_SERIES_MODULE, ProductCategory.SCXI_MODULE]:
        return f"/{device.name}/{terminal_name}"
```

Reply:
The suggested codes work, thank you. I have added it in the examples.